File: aula-extra-est-main/FrontendAulaExtra/AulaExtra/lib/core/data/student_evaluations/dtos/pending_evaluation_dto.dart
class PendingEvaluationDto {
  PendingEvaluationDto({
    required this.lessonId,
    required this.professorId,
    required this.professorName,
    required this.subject,
    required this.lessonStart,
    required this.lessonEnd,
  });

  final String lessonId;
  final String professorId;
  final String professorName;
  final String? subject;
  final DateTime? lessonStart;
  final DateTime? lessonEnd;

  factory PendingEvaluationDto.fromJson(Map<String, dynamic> json) {
    return PendingEvaluationDto(
      lessonId: (json['lessonId'] ?? '').toString(),
      professorId: (json['professorId'] ?? '').toString(),
      professorName: (json['professorName'] ?? '').toString(),
      subject: json['subject']?.toString(),
      lessonStart: _tryParseDateTime(json['lessonStart']),
      lessonEnd: _tryParseDateTime(json['lessonEnd']),
    );
  }

  static DateTime? _tryParseDateTime(dynamic value) {
    if (value == null) return null;
    final s = value.toString();
    if (s.trim().isEmpty) return null;
    return DateTime.tryParse(s);
  }
}

File: aula-extra-est-main/BackendAulaExtra/src/Modules/Student/DTOs/PendingEvaluationDto.cs
using System;

namespace ConfidantPostgreSQL.Modules.Student.DTOs
{
    public class PendingEvaluationDto
    {
        public Guid LessonId { get; set; }
        public Guid ProfessorId { get; set; }
        public string ProfessorName { get; set; } = string.Empty;
        public string? Subject { get; set; }
        public DateTime? LessonStart { get; set; }
        public DateTime? LessonEnd { get; set; }
    }
}
